### PR TITLE
foam3 user to subject.user transfer

### PR DESCRIPTION
**Pull request by microArtur**
https://nanopay.atlassian.net/browse/NP-1363

### CODE REVIEW

File: src/foam/dao/history/HistoryRecord.js [hunk 1]
```diff
@@ -37,9 +37,9 @@ foam.CLASS({
       name: 'subject',
       documentation: `References subject that made update. TODO: remove references when subject becomes serializable`,
       postSet: function(o, n) {
-        this.userId = n.user.id;
+        this.userId = n.user.id; // this line?
         this.agentId = n.realUser.id;
-        this.user = n.user.toSummary();
+        this.subject.user = n.user.toSummary();
```
Comment (smirnovaka) [inline src/foam/dao/history/HistoryRecord.js]: i think you shouldn't change this line. This updates the local(instance class) variable

File: src/foam/nanos/auth/ProfilePictureView.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ foam.CLASS({
 
   imports: [
     'user',
```
Comment (smirnovaka) [inline src/foam/nanos/auth/ProfilePictureView.js]: might as well remove user import if not used

File: src/foam/nanos/crunch/test/CrunchTestSupport.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ foam.CLASS({
 
   properties: [
     {
-      name: 'user',
+      name: 'user',  // is a property here
```
Comment (smirnovaka) [inline src/foam/nanos/crunch/test/CrunchTestSupport.js]: do we need this comment?

File: src/foam/nanos/dig/DigFileUploadView.js [hunk 4]
```diff
@@ -20,6 +20,7 @@ foam.CLASS({
 
   imports: [
     'user',
+    'subject',
```
Comment (smirnovaka) [inline src/foam/nanos/dig/DigFileUploadView.js:22]: remove user if not used? here and everywhere else

File: src/foam/nanos/u2/navigation/SignIn.js [hunk 5]
```diff
@@ -141,23 +141,25 @@ foam.CLASS({
...
-                      err: err.data,
-                      message: this.ERROR_MSG,
-                      type: this.LogLevel.ERROR
-                    }));
-                  });
-              } else {
-                this.user.copyFrom(logedInUser);
-                this.nextStep();
-              }
+                if ( this.token_ ) {
```
Comment (smirnovaka) [inline src/foam/nanos/u2/navigation/SignIn.js]: @zahnno could you please review this file

Comment (zahnno) [reply src/foam/nanos/u2/navigation/SignIn.js]: lgtm

File: src/foam/dao/history/HistoryRecord.js (hunk 1 above)
Comment (lchanmann) [inline src/foam/dao/history/HistoryRecord.js]: why assigning user summary to the subject.user object?

File: src/foam/nanos/auth/UpdatePassword.js [hunk 6]
```diff
@@ -165,7 +165,7 @@ foam.CLASS({
       code: function() {
         this.auth.updatePassword(null, this.originalPassword, this.newPassword)
         .then((result) => {
-          this.user.copyFrom(result);
+          this.subject.user.copyFrom(result);
```
Comment (lchanmann) [inline src/foam/nanos/auth/UpdatePassword.js]: if the auth.updatePassword() only changes the user password, might not need to copy back to the subject.user on the client side as the password field is networkTransient.

File: src/foam/dao/history/HistoryRecord.js (hunk 1 above)
Comment (microArtur) [reply src/foam/dao/history/HistoryRecord.js]: thought that changes required here, but user is private variable in this case, so no changes required

File: src/foam/nanos/u2/navigation/SignIn.js (hunk 5 above)
Comment (lchanmann) [inline src/foam/nanos/u2/navigation/SignIn.js]: don't need the indentation as it's not a nested if, can add a blank line between line 143 and 144.

Comment (microArtur) [reply src/foam/nanos/u2/navigation/SignIn.js]: not sure why this happen

**Review by smirnovaka — APPROVED** [review]

**Pull request merged**